jetty-http3/http3-qpack/src/main/java/org/eclipse/jetty/http3/qpack/QpackEncoder.java

```diff
@@ -131,11 +162,16 @@ public int getCapacity()
      *
      * @param capacity the new capacity.
      */
-    public void setCapacity(int capacity)
+    public void setTableCapacity(int capacity)
     {
-        _context.getDynamicTable().setCapacity(capacity);
-        _handler.onInstructions(List.of(new SetCapacityInstruction(capacity)));
-        notifyInstructionHandler();
+        try (AutoLock ignored = lock.lock())
```

Comment:
I didn't even know there was a lock in this class. When I was writing it you said specifically not to have locks and any synchronisation will be done externally.

But since its here now you will also need to use it inside `QPackEncoder.streamCancellation` and `QPackEncoder.insert`.

Reply:
The decoder can be lock free because reads are serialized, and it only reads.

But for the encoder, it can called by the application (write-like operation) at the same time it is reading instructions, so there is the need for internal locking.

Added locking to `streamCancellation()` and `insert()` too, although the latter is only used for testing.

Reply:
I think we could possibly be a lot more efficient with the locks and lock only when we need to make a modification to the table.

But not something for this PR anyway.